app/views/govuk_publishing_components/components/docs/accordion.yml

```diff
@@ -274,6 +274,32 @@ examples:
             text: "How people read"
           content:
             html: "<p class='govuk-body'>This is the content for How people read.</p>"
+  with_id_on_headings:
+    description: |
+      Some apps require custom ids per accordion section heading for linking between those specific sections across multiple pages. An example of this is on manuals pages where multiple manuals will each include large sets of accordions and will reference between specific sections for ease of access to that content.
+
+      This attribute is paired with functionality to automatically open an accordion section on page load where the URL hash is the same as the id of that accordion. [Live example](https://www.gov.uk/guidance/how-to-publish-on-gov-uk/creating-and-updating-pages#associations)
+
+      Unlike with the accordion-wide custom id attribute, this id isn't stored in `localStorage` so does not need to be unique across your domain, but should still be unique to the page.
```

Comment:
This is great to mention this **unique** aspect. I'm wondering if this should be thought about when it comes to the API work.  

I'm thinking about the easy of duplicate IDs happening..

Reply:
Rails has a thing for generating random values that are unique called `SecureRandom`. I don't think it'd be useful here (looks like if no id is passed no id is created) but thought it worth mentioning.

Reply:
Ah thanks @andysellick, good to know, To give _some_ context in a publishing app an ID might be passed to link elsewhere - eg

`## Add associations (tagging) 
{:#associations}`

Reply:
Remember that there's a lot we still don't know. Manuals publisher might have validation for duplicate ids. This wouldn't surprise me to be honest, it's a feature in Whitehall of all places! I don't think policing data here is a valuable use of time.